src/ast-utils/feature-finder.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+// jshint node: true
+'use strict';
+
+import * as estraverse from 'estraverse';
+import * as esutil from './esutil';
+import {FeatureDescriptor} from './descriptors';
+import {Visitor} from './fluent-traverse';
+import * as estree from 'estree';
+
+var numFeatures = 0;
+
+export function featureFinder() {
+  /** The features we've found. */
+  var features : FeatureDescriptor[] = [];
```

Comment:
what's our style with colons separating type annotations? I actually prefer this style, as you know :)